client-runtime/src/main/java/com/microsoft/rest/v2/annotations/QueryParam.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.rest.v2.annotations;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Retention(RUNTIME)
+@Target(PARAMETER)
+public @interface QueryParam {
+    /**
+     * The name of the variable in the endpoint uri template which will be replaced with the value
+     * of the parameter annotated with this annotation.
+     */
+    String value();
+    /**
+     * A value true for this argument indicates that value of {@link QueryParam#value()} is already encoded
+     * hence engine should not encode it, by default value will be encoded.
+     */
```

Comment:
Consider this: "This value indicates whether or not this parameter's value is encoded. If the parameter's value is encoded, the engine will not attempt to encode it again. The default value is false, meaning the engine will attempt to encode the parameter's value."